### PR TITLE
Adds documentation for Download

### PR DESCRIPTION
**Pull request by kul3r4**
#### WHAT
Adds documentation for downloading media.

#### WHY
Current documentation does not explain how progress is implemented and which type of UI screen we provide.

#### HOW
Provide some code snippets to give context.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: docs/media-data.md [hunk 1]
```diff
@@ -28,3 +28,176 @@ internal database.
 ```
 
 [media3]: https://developer.android.com/jetpack/androidx/releases/media3
+
+### Media Toolkit implementation
+
+#### DownloadManagerListener
+The `DownloadManagerListener` is an implementation of listener for [DownloadManager][downloadmanager] events which can also get notified of
```
Comment (yschimke) [inline docs/media-data.md:35]: Maybe worth calling out it's Media3 DownloadManager and DownloadService.

File: docs/media-data.md [hunk 2]
```diff
@@ -28,3 +28,176 @@ internal database.
...
+override fun onDownloadRemoved(downloadManager: DownloadManager, download: Download) {
+        coroutineScope.launch {
+            val mediaId = download.request.id
+            mediaDownloadLocalDataSource.delete(mediaId)
+        }
+    }
+```
+
+#### DownloadProgressMonitor
+The `DownloadProgressMonitor` monitors the status of the download by polling the `DownloadManager`
```
Comment (yschimke) [inline docs/media-data.md:67]: We should probably highlight that we duplicate the status from Media3's StandaloneDatabaseProvider, but that isn't available for our UI in an easy to query or reactive way.

Unrelated to this PR, we should probably suggest a modern Room implementation of the  StandaloneDatabaseProvider to the media team.

**Review by yschimke — APPROVED** [review]

File: docs/media-data.md (hunk 1 above)
Comment (kul3r4) [reply docs/media-data.md:35]: If my syntax is working well, this is linked to the Media 3 resources, see line 202 of this file

Comment (yschimke) [reply docs/media-data.md:35]: Yep, the links work, but are only visible after clicking. Your call.

Comment (kul3r4) [reply docs/media-data.md:35]: done, thanks

**Pull request merged**